types/src/transaction/use_case.rs

```diff
@@ -52,3 +55,46 @@ impl UseCaseAwareTransaction for SignedTransaction {
         }
     }
 }
+
+impl UseCaseAwareTransaction for SignatureVerifiedTransaction {
+    fn parse_sender(&self) -> AccountAddress {
+        self.sender()
+            .expect("Expected a sender on SignatureVerifiedTransaction but received None")
+    }
+
+    fn parse_use_case(&self) -> UseCaseKey {
+        use crate::transaction::TransactionPayload::*;
+        use UseCaseKey::*;
+
+        let payload: Option<&TransactionPayload> = match self {
+            SignatureVerifiedTransaction::Valid(txn) => match txn {
+                Transaction::UserTransaction(signed_txn) => Some(signed_txn.payload()),
+                Transaction::GenesisTransaction(_)
+                | Transaction::BlockMetadata(_)
+                | Transaction::StateCheckpoint(_)
+                | Transaction::ValidatorTransaction(_)
+                | Transaction::BlockMetadataExt(_)
+                | Transaction::BlockEpilogue(_) => None,
+            },
+            // TODO I don't think we want invalid transactions during shuffling, but double check this logic...
+            SignatureVerifiedTransaction::Invalid(_) => None,
+        };
+
+        match payload {
```

Comment:
nit: if you like, remove nested matching by
```
let payload = payload.expect("...");
match payload {
   Script(_) | ...
}

```

Reply:
Great suggestion, thanks!